test/integration/schema/index.js

```diff
@@ -576,5 +576,111 @@ module.exports = function(knex) {
       });
     });
 
+    it('supports named primary keys', function() {
+      var constraintName = 'pk-test';
+      var tableName = 'namedpk';
+      return knex.transaction(function(tr) {
+        return tr.schema.dropTableIfExists(tableName)
+          .then(function() {
+            return tr.schema.createTable(tableName, function(table) {
+              table.string('test').primary(constraintName);
+            })
+          })
+          .then(function() {
+            return tr.schema.dropTableIfExists(tableName);
+          })
+          .then(function() {
+            return tr.schema.createTable(tableName, function(table) {
+              table.string('test');
+              table.string('test2');
+            })
+          })
+          .then(function() {
+            return tr.schema.table(tableName, function(table) {
+              table.primary('test', constraintName)
+            })
+          })
+          .then(function() {
+            return tr.schema.dropTableIfExists(tableName);
```

Comment:
Here you should try to drop constraint with raw query or do something else to make sure that given name was used for creating the named constraint.